### PR TITLE
feat: New Module `avm/res/cdn/profile`

### PR DESCRIPTION
**Pull request by krbar**
## Description

New Module `avm/res/cdn/profile`, migrated from CARML.

## Adding a new module

- [x] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Pipeline references

| Pipeline |
| - |
| [avm.res.cdn.profile](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.cdn.profile.yml) |

### CODE REVIEW

File: avm/res/cdn/profile/afdEndpoint/main.bicep [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+param enabledState string = 'Enabled'
+
+@description('Optional. The list of routes for this AFD Endpoint.')
+param routes array?
+
+resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
+  name: profileName
+}
+
+resource afd_endpoint 'Microsoft.Cdn/profiles/afdEndpoints@2023-05-01' = {
```
Comment (AlexanderSehr) [inline avm/res/cdn/profile/afdEndpoint/main.bicep:40]: Would suggest to use `afdEndpoint` as it is more consistent with the rest of the naming

File: avm/res/cdn/profile/customdomain/main.bicep [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+@description('Optional. The minimum TLS version required for the custom domain. Default value: TLS12.')
+param minimumTlsVersion string = 'TLS12'
+
+@description('Optional. The name of the secret. ie. subs/rg/profile/secret.')
+param secretName string = ''
+
+resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
+  name: profileName
+
+  resource profile_secrect 'secrets@2023-05-01' existing = if (!empty(secretName)) {
```
Comment (AlexanderSehr) [inline avm/res/cdn/profile/customdomain/main.bicep:43]:
```suggestion
  resource secrect 'secrets@2023-05-01' existing = if (!empty(secretName)) {
```

File: avm/res/cdn/profile/customdomain/main.bicep [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    name: secretName
+  }
+}
+
+resource profile_custom_domain 'Microsoft.Cdn/profiles/customDomains@2023-05-01' = {
```
Comment (AlexanderSehr) [inline avm/res/cdn/profile/customdomain/main.bicep:48]: Should be `customDomain`

File: avm/res/cdn/profile/endpoint/main.bicep [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  params: {
+    profileName: profile.name
+    endpointName: endpoint.name
+    name: origin.name
+    hostName: origin.properties.hostName
+    httpPort: origin.properties.?httpPort
+    httpsPort: origin.properties.?httpsPort
+    enabled: origin.properties.enabled
+    priority: origin.properties.?priority ?? -1
+    weight: origin.properties.?weight ?? -1
```
Comment (AlexanderSehr) [inline avm/res/cdn/profile/endpoint/main.bicep:43]:
```suggestion
    priority: origin.properties.?priority
    weight: origin.properties.?weight
```

Should pick the default automatically

File: avm/res/cdn/profile/main.bicep [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+  name: '${uniqueString(deployment().name, location)}-Profile-Endpoint'
+  params: {
+    name: endpointName ?? '${profile.name}-endpoint'
+    properties: endpointProperties ?? {}
+    location: location
+    profileName: profile.name
+  }
+}
+
+module profile_secret 'secret/main.bicep' = [for (secret, index) in secrets: {
```
Comment (AlexanderSehr) [inline avm/res/cdn/profile/main.bicep:140]:
```suggestion
module profile_secrets 'secret/main.bicep' = [for (secret, index) in secrets: {
```

File: avm/res/cdn/profile/customdomain/main.bicep (hunk 3 above)
Comment (krbar) [reply avm/res/cdn/profile/customdomain/main.bicep]: ok: https://github.com/Azure/bicep-registry-modules/pull/1106/commits/264d035f36513fe6c268ee0c65fc01340ca5eacd

File: avm/res/cdn/profile/afdEndpoint/main.bicep (hunk 1 above)
Comment (krbar) [reply avm/res/cdn/profile/afdEndpoint/main.bicep]: yep: https://github.com/Azure/bicep-registry-modules/pull/1106/commits/264d035f36513fe6c268ee0c65fc01340ca5eacd

**Comment by Cadacious** [comment]
Hello, is is possible to cover the [Microsoft.Cdn/cdnWebApplicationFirewallPolicies](https://learn.microsoft.com/en-us/azure/templates/microsoft.cdn/cdnwebapplicationfirewallpolicies?pivots=deployment-language-bicep) resource under this PR as well please? 

This is commonly deployed with the `Microsoft.Cdn` based front door.

**Comment by AlexanderSehr** [comment]
> Hello, is it possible to cover the [Microsoft.Cdn/cdnWebApplicationFirewallPolicies](https://learn.microsoft.com/en-us/azure/templates/microsoft.cdn/cdnwebapplicationfirewallpolicies?pivots=deployment-language-bicep) resource under this PR as well please?
> 
> This is commonly deployed with the `Microsoft.Cdn` based front door.

Hey @Cadacious,

thanks for bringing this up. This PR specifically focuses on migrating the original CARML module to AVM, so that we can retire it from there. Any chance you could open an issue via this [link](https://github.com/Azure/Azure-Verified-Modules/issues/new?assignees=&labels=Type%3A+New+Module+Proposal+%3Abulb%3A%2CNeeds%3A+Triage+%3Amag%3A&projects=Azure%2F529&template=module_proposal.yml&title=%5BModule+Proposal%5D%3A+%60MODULE_NAME_BETWEEN_BACKTICKS%60) in AVM so track the module request? :)

File: avm/res/cdn/profile/main.bicep [hunk 6]
```diff
@@ -127,7 +127,7 @@ resource profile_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-0
   scope: profile
 }]
 
-module profile_endpoint 'endpoint/main.bicep' = if (!empty(endpointProperties)) {
+module profile_endpoints 'endpoint/main.bicep' = if (!empty(endpointProperties)) {
```
Comment (AlexanderSehr) [inline avm/res/cdn/profile/main.bicep:130]:
```suggestion
module profile_endpoint 'endpoint/main.bicep' = if (!empty(endpointProperties)) {
```

That one is actually only one 😄

File: avm/res/cdn/profile/tests/e2e/afd/main.test.bicep [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    storageAccountName: 'dep${namePrefix}cdnstore${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+    location: resourceLocation
+  }
+}
```
Comment (AlexanderSehr) [inline avm/res/cdn/profile/tests/e2e/afd/main.test.bicep:42]: It appears non of the dependencies are used?

File: avm/res/cdn/profile/tests/e2e/waf-aligned/main.test.bicep [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    storageAccountName: 'dep${namePrefix}cdnstore${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
```
Comment (AlexanderSehr) [inline avm/res/cdn/profile/tests/e2e/waf-aligned/main.test.bicep:39]: Isn't used as far as I can see

File: avm/res/cdn/profile/tests/e2e/afd/main.test.bicep (hunk 7 above)
Comment (krbar) [reply avm/res/cdn/profile/tests/e2e/afd/main.test.bicep]: good catch, thanks. removed: https://github.com/Azure/bicep-registry-modules/pull/1106/commits/8fe70c4ad3099cbb6a612875d190650e85134319

File: avm/res/cdn/profile/tests/e2e/waf-aligned/main.test.bicep (hunk 8 above)
Comment (krbar) [reply avm/res/cdn/profile/tests/e2e/waf-aligned/main.test.bicep]: Managed identity removed: https://github.com/Azure/bicep-registry-modules/pull/1106/commits/8fe70c4ad3099cbb6a612875d190650e85134319

**Comment by krbar** [comment]
@AlexanderSehr thanks for the review, I think all the comments are addressed now.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**